### PR TITLE
MGMT-11925: Align common workflow with the old baremetal step

### PR DESCRIPTION
**Pull request by adriengentil**
Few variables were not set properly, also with the common workflow, the old `MAKEFILE_TARGET` was split in 2:
- `MAKEFILE_SETUP_TARGET` is run in `assisted-common-setup-prepare`
  step
- `MAKEFILE_TARGET` is run in `assisted-common-test` step

### CODE REVIEW

File: ci-operator/step-registry/assisted/common/setup/prepare/assisted-common-setup-prepare-commands.sh [hunk 1]
```diff
@@ -101,15 +103,15 @@ export MAKEFILE_TARGET='setup test_parallel'
...
+{% if HYPERSHIFT_IMAGE != ASSISTED_CONTROLLER_IMAGE %}
 export HYPERSHIFT_IMAGE={{ HYPERSHIFT_IMAGE }}
 {% endif %}
 
 {% if JOB_TYPE == "presubmit" and REPO_NAME == "assisted-service" %}
 export SERVICE_BRANCH={{ PULL_PULL_SHA }}
 {% endif %}
 
-{% if REPO_NAME != "assisted-service" %}
+{% if JOB_TYPE != "presubmit" or  REPO_NAME == "release" %}
```
Comment (osherdp) [inline ci-operator/step-registry/assisted/common/setup/prepare/assisted-common-setup-prepare-commands.sh:114]: nit: an extra space in there

btw, I do not recall what we originally intended with this code, but with this change (and with the previous code) we are having different images in assisted-service presubmit jobs and in rehearsals of those jobs
Are we able to fix it in some way? (maybe there's some reference to which repo we're rehearsing in the prow env?)

**Review by osherdp — CHANGES_REQUESTED** [review]

File: ci-operator/step-registry/assisted/common/setup/prepare/assisted-common-setup-prepare-commands.sh (hunk 1 above)
Comment (adriengentil) [reply ci-operator/step-registry/assisted/common/setup/prepare/assisted-common-setup-prepare-commands.sh:114]: :eagle:

It was here in the old script: https://github.com/openshift/release/blob/e6651ea1fbbb43e98c294b39098aa2d407680cd1/ci-operator/step-registry/assisted/baremetal/setup/assisted-baremetal-setup-commands.sh#L146-L158

and you approved https://github.com/openshift/release/pull/18291 :P

Comment (adriengentil) [reply ci-operator/step-registry/assisted/common/setup/prepare/assisted-common-setup-prepare-commands.sh:114]: but, yes I don't understand why we do that

File: ci-operator/config/openshift/assisted-test-infra/openshift-assisted-test-infra-release-ocm-2.4.yaml [hunk 2]
```diff
@@ -109,7 +109,7 @@ tests:
         ISO_IMAGE_TYPE=minimal-iso
         KUBE_API=yes
         ENABLE_KUBE_API=true
-        MAKEFILE_TARGET="create_full_environment run test_kube_api_parallel"
+        MAKEFILE_TARGET="test_kube_api_parallel"
         TEST_FUNC=test_kubeapi
```
[the remark this answers is not in the document]
Comment (adriengentil) [reply ci-operator/config/openshift/assisted-test-infra/openshift-assisted-test-infra-release-ocm-2.4.yaml:113]: [create_full_environment](https://github.com/openshift/assisted-test-infra/blob/release-ocm-2.4/Makefile#L169) is an alias for `setup`, so I guess it's fine? Or do you mean something else?

File: ci-operator/step-registry/assisted/common/setup/prepare/assisted-common-setup-prepare-commands.sh (hunk 1 above)
Comment (adriengentil) [reply ci-operator/step-registry/assisted/common/setup/prepare/assisted-common-setup-prepare-commands.sh:114]: may be let's keep this problem out of this PR, I will open a ticket to address the issue separately. I would first like to fix the failing capi jobs.

File: ci-operator/config/openshift/assisted-test-infra/openshift-assisted-test-infra-release-ocm-2.4.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (osherdp) [reply ci-operator/config/openshift/assisted-test-infra/openshift-assisted-test-infra-release-ocm-2.4.yaml:113]: I mean the usage of ``deploy_nodes_with_install`` versus ``test_kube_api_parallel``

[the remark this answers is not in the document]
Comment (adriengentil) [reply ci-operator/config/openshift/assisted-test-infra/openshift-assisted-test-infra-release-ocm-2.4.yaml:113]: oh yeah, `make all` does `setup`, `run` and `deploy_nodes_with_install`, I just adapted. But yes, that's wierd, it should be `test_kube_api_parallel` I guess.

File: ci-operator/step-registry/assisted/common/setup/prepare/assisted-common-setup-prepare-commands.sh (hunk 1 above)
Comment (osherdp) [reply ci-operator/step-registry/assisted/common/setup/prepare/assisted-common-setup-prepare-commands.sh:114]: sure no problem
thanks!

**Review by osherdp — APPROVED** [review]

**Review by osherdp — APPROVED** [review]

**Review by osherdp — APPROVED** [review]

**Comment by adriengentil** [comment]
@osherdp I think we are good to go (only one failure due to equinix provisioning)

**Review by osherdp — APPROVED** [review]

**Pull request merged**